clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -2496,6 +2874,55 @@ check_read_write_records_match_operands()
     return true;
 }
 
+bool
+check_exit_found(void)
+{
+    std::cerr << "Testing thread exits\n";
+    // Correct: all threads have exits.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_marker(TID_A, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+            gen_instr(TID_A),
+            gen_exit(TID_A),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+            gen_instr(TID_B),
+            gen_exit(TID_B),
+            gen_marker(TID_C, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64),
+            gen_marker(TID_C, TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+            gen_instr(TID_C),
+            gen_exit(TID_C),
+        };
+        if (!run_checker(memrefs, false))
+            return false;
+    }
+    // Incorrect: a thread is missing an exit.
```

Comment:
nit: thread B is missing an exit?

Reply:
Yes but that is the goal!  As the comment here and above say, this is deliberately missing an exit.